app/portainer/views/account/CreateAccessToken/CreateAccessToken.tsx

```diff
@@ -0,0 +1,112 @@
+import { PropsWithChildren, useEffect, useState } from 'react';
+
+import { Widget, WidgetBody } from '@/portainer/components/widget';
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { TextInput } from '@/portainer/components/form-components/Input';
+import { Button } from '@/portainer/components/Button';
+import { FormSectionTitle } from '@/portainer/components/form-components/FormSectionTitle';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+import { Code } from '@/portainer/components/Code';
+import { CopyButton } from '@/portainer/components/Button/CopyButton';
+
+import styles from './CreateAccessToken.module.css';
+
+interface AccessTokenResponse {
+  rawAPIKey: string;
+}
+
+export interface Props {
+  // userId for whom the access token is generated for
+  userId: number;
+
+  // onSubmit dispatches a successful matomo analytics event
+  onSubmit: (description: string) => Promise<AccessTokenResponse>;
+
+  // onSuccess is called when upon clicking the done button
+  onSuccess: () => void;
+
+  // onError is called when an error occurs; this is a callback to Notifications.error
+  onError: (heading: string, err: unknown, message: string) => void;
+}
+
+export function CreateAccessToken({
+  onSubmit,
+  onSuccess,
+  onError,
+}: PropsWithChildren<Props>) {
+  const [description, setDescription] = useState('');
+  const [errorText, setErrorText] = useState<string>('');
```

Comment:
like in description, the generic type is inferred, we don't need to state it if the default value is provided